spec.html

```diff
@@ -3042,6 +3095,25 @@ <h1>CheckBoundNames ( _expectedBound_, _actualBound_ )</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-checkinnerlazyfunctions" aoid="CheckInnerLazyFunctions">
+      <h1>CheckInnerLazyFunctions ( _funcNode_ )</h1>
+      <emu-alg>
+          1. Assert: _funcNode_ is a `LazyFunctionDeclaration`, a `LazyFunctionExpression`, a `LazyMethod`, a `LazyGetter`, a `LazySetter`, or a `LazyArrowExpression`.
+          1. NOTE: All asserted immediately inner (i.e. not nested within another inner function) lazy functions in `innerLazyFunctions` must be found.
+          1. Let _innerFunctions_ be a new empty List.
+          1. For each _link_ in _funcNode_`.innerLazyFunctions`, do
+              1. Let _linkedNode_ be the node linked to by _link_.
```

Comment:
shouldn't we assert that the _linkedNode_ is direct inner function of _funcNode_?

Reply:
Like assert that it is not nested inside another inner function? That's supposed to be taken care of by the very step: if _innerFunctions_ isn't empty, then we throw.

Reply:
does it mean that such link is allowed per file format?
and even if such link appears in the file, we should check if each entry has corresponding functions and throw SyntaxError for it if there's not?

Reply:
We'll certainly know when delazifying if there's a Linked* that doesn't exist, and we can throw then, before exer executing anything.

Is there something in particular that has you worried?

Reply:
I'm thinking about the order of the error.
like, if there's 2 possible errors, which one should be observed?

maybe I should revisit this (and some other errors) once the surface encoding format is fixed.